packages/cfd/src/Components/__tests__/cfd-dxtrade-demo-account-display.spec.tsx

```diff
@@ -0,0 +1,74 @@
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import CFDDxtradeDemoAccountDisplay from '../cfd-dxtrade-demo-account-display';
+import { TCFDPlatform } from '../props.types';
+
+const mock_connect_props = {
+    dxtrade_tokens: {
+        demo: '',
+        real: '',
+    },
+};
+
+jest.mock('Stores/connect.js', () => ({
+    __esModule: true,
+    default: 'mockedDefaultExport',
+    connect:
+        () =>
+        <T extends React.FC>(Component: T) =>
+        (props: React.ComponentProps<T>) =>
+            Component({ ...props, ...mock_connect_props }),
+}));
+
+const mock_props = {
+    has_cfd_account_error: false,
+    standpoint: {
+        financial_company: 'svg',
+        gaming_company: 'svg',
+        iom: false,
+        malta: false,
+        maltainvest: false,
+        svg: true,
+    },
+    is_loading: false,
+    is_logged_in: true,
+    onSelectAccount: jest.fn(),
+    openAccountTransfer: jest.fn(),
+    platform: 'dxtrade' as TCFDPlatform,
+    current_list: { 'dxtrade.demo.dxtrade': { enabled: 1 } },
+    openPasswordManager: jest.fn(),
+};
+
+describe('CFDDxtradeDemoAccountDisplay', () => {
+    it('should render Loading component if is_loading === true', () => {
+        render(<CFDDxtradeDemoAccountDisplay {...mock_props} is_loading />);
+
+        expect(screen.getByTestId('dt_barspinner')).toBeInTheDocument();
+    });
+    it('should render CFDAccountCard component if is_loading === false', () => {
+        render(<CFDDxtradeDemoAccountDisplay {...mock_props} />);
+
+        expect(screen.getByText('Deriv X')).toBeInTheDocument();
+    });
+    it('should render CFDAccountCard with Top Up button if existing_accounts_data fucntion does not return undefined', () => {
+        render(<CFDDxtradeDemoAccountDisplay {...mock_props} />);
+
+        expect(screen.getByText('Top up')).toBeInTheDocument();
```

Comment:
@kate-deriv 😊 nit: since we use 'Top up' string multiple times, let's put it into a variable and reuse it👍🏻

Reply:
Sure, good idea

Reply:
But if I put it in a variable, then the component should be rendered before it (because the button won't exist earlier). And here comes the problem: if I lift up render in `beforeEach`, then it isn't possible to pass different `mock_props` in it. E.g. in 1st and 4th I pass not default props.

Reply:
that's why I suggest declaring a variable for the string only, like this: `const top_up_button_title = 'Top up';` 🙂

Reply:
Oh, I see what you mean, no problem)